WebHostLib/misc.py

```diff
@@ -37,23 +40,25 @@ def start_playing():
     return render_template(f"startPlaying.html")
 
 
-# TODO for back compat. remove around 0.4.5
-@app.route("/weighted-settings")
-def weighted_settings():
-    return redirect("weighted-options", 301)
-
-
-@app.route("/weighted-options")
-@cache.cached()
-def weighted_options():
-    return render_template("weighted-options.html")
-
-
-# Player options pages
-@app.route("/games/<string:game>/player-options")
+@app.route("/games/<string:game>/option-presets", methods=["GET"])
 @cache.cached()
-def player_options(game: str):
-    return render_template("player-options.html", game=game, theme=get_world_theme(game))
+def option_presets(game):
+    world = AutoWorldRegister.world_types[game]
+    presets = {}
+
+    if world.web.options_presets:
+        presets = presets | world.web.options_presets
+
+    class SetEncoder(json.JSONEncoder):
+        def default(self, obj):
+            if isinstance(obj, set):
```

Comment:
For some parts of `Options` we use `frozenset` as a default (because we want immutable defaults).
But this doesn't handle `frozenset`
```python
>>> fs = frozenset()
>>> json.dumps(fs)
Traceback (most recent call last):
...
TypeError: Object of type frozenset is not JSON serializable
>>> isinstance(fs, set)
False
>>> from collections.abc import Set             
>>> isinstance(fs, Set)
True
>>> isinstance(set(), Set)
True
```

```suggestion
            from collections.abc import Set
            if isinstance(obj, Set):
```
`collections.abc.Set` will catch both `set` and `frozenset`

Reply:
I implemented your suggested change to `collections.abs.Set`.